exotations/exotica_core_task_maps/src/manipulability.cpp

```diff
@@ -33,23 +33,22 @@ REGISTER_TASKMAP_TYPE("Manipulability", exotica::Manipulability);
 
 namespace exotica
 {
-Manipulability::Manipulability() = default;
-Manipulability::~Manipulability() = default;
-
 void Manipulability::Update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi)
 {
     if (phi.rows() != TaskSpaceDim()) ThrowNamed("Wrong size of phi!");
 
     for (int i = 0; i < n_end_effs_; ++i)
     {
         const Eigen::MatrixXd& J = kinematics[0].jacobian[i].data.topRows(n_rows_of_jac_);
-        phi(i) = std::sqrt((J * J.transpose()).determinant());
+        phi(i) = -std::sqrt((J * J.transpose()).determinant());
```

Comment:
This does not correspond to the description in `exotica_plugins` above

Reply:
I.e., you changed the sign here - but say above (in newly added comment) that the Rho needs to be negative to maximise. Is the minus required for the inequality to work?

Reply:
I have updated `exotica_plugins` to be a little more descriptive.

Let `m = sqrt(det(J' * J))`. Above, `Goal` was requested to be used as the lower bound when the task map is used as an inequality constraint so `Phi` needs to take the form `Phi = -m` in order to satisfy both being set as a cost term and inequality. 
As a cost term, the objective term is `Rho * ydiff' * ydiff` where `ydiff = Phi - Goal = -m - Zeros`. To maximize the manipulability here `Rho` must be negative and the negative sign is squared so doesn't affect outcome.
As an inequality, the goal is `m >= LB <=> -m + LB <= 0` where `LB>0` is some strictly positive lower bound. For inequalities exotica specifies `ydiff = Phi - Goal <= 0`. Thus, in-order to achieve the desired inequality whilst corresponding with the cost-version of the task map and not introducing the `lower_bound_` parameter, `Phi` must be `-m` and `Goal<0`must take strictly negative values.

Reply:
What might make sense is to perform a check in the `Update` that ensures `Goal` is strictly negative or zero. I'm not sure that it's possible to access `Goal` from within the task map?

Reply:
You can't, since that is _task_ and not _task-map_ dependent.

Reply:
Ah of course, cheers.